src/components/Proposal.tsx

```diff
@@ -40,7 +40,7 @@ class DAOProposal extends Component<Props, Entity, Data, Code>
       throw Error("DAO Missing: Please provide this field as a prop, or use the inference component.");
     }
 
-    return dao.proposal(id);
+    return new Entity(id, dao.address, dao.context);
```

Comment:
Why this instead of dao.proposal.search({}) or dao.proposal(id) ?

Reply:
I want to start only relying on the Entity constructor for regular components, and using the Entity.search({}) method for component lists. This way it's standardized across all components, and once we have string templates to generate these component.tsx files it'll make that easier.